src/main/java/com/example/inventoryservice/service/impl/IngredientServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Service
+public class IngredientServiceImpl implements IngredientService {
+    private final IngredientRepository ingredientRepository;
+    private final IngredientConverter ingredientConverter;
+    private final UserService userService;
+    Logger logger = LoggerFactory.getLogger(UserServiceImpl.class);
+
+    public IngredientServiceImpl(IngredientRepository ingredientRepository, IngredientConverter ingredientConverter, UserService userService) {
+        this.ingredientRepository = ingredientRepository;
+        this.ingredientConverter = ingredientConverter;
+        this.userService = userService;
+    }
+
+    @Override
+    public IngredientDto create(IngredientDto ingredientDto) {
+        logger.info("Create ingredient");
+        UserDto userDto = userService.getCurrentUser();
+        RestaurantDto restaurantDto = userDto.getRestaurant();
+        ingredientDto.setRestaurant(restaurantDto);
+        Ingredient ingredient = ingredientConverter.dtoToEntity(ingredientDto);
+        if (ingredientExists(ingredient)) {
+            throw new ServiceException("Not unique ingredient");
+        }
+        Ingredient persistIngredient = ingredientRepository.save(ingredient);
+        return ingredientConverter.entityToDto(persistIngredient);
+    }
+
+    @Override
+    public IngredientDto updateAmount(IngredientDto ingredientDto) {
+        logger.info("Update ingredient amount");
+        Ingredient ingredient = ingredientConverter.dtoToEntity(ingredientDto);
+        if (!ingredientExists(ingredient)) {
+            logger.error("No such ingredient {}", ingredient.getName());
+            throw new ServiceException("No such ingredient");
+        }
+        Ingredient persistIngredient = ingredientRepository.findById(ingredient.getId())
+                                                           .orElse(new Ingredient());
+        BigDecimal persistAmount = persistIngredient.getAmount();
+        persistIngredient.setAmount(persistAmount.add(ingredient.getAmount()));
+        persistIngredient = ingredientRepository.save(persistIngredient);
+        return ingredientConverter.entityToDto(persistIngredient);
+    }
+
+    @Override
+    public IngredientDto updatePrice(IngredientDto ingredientDto) {
+        logger.info("Update ingredient price");
+        Ingredient ingredient = ingredientConverter.dtoToEntity(ingredientDto);
+        if (!ingredientExists(ingredient)) {
+            logger.error("No such ingredient {}", ingredient.getName());
+            throw new ServiceException("No such ingredient");
+        }
+        Ingredient persistIngredient = ingredientRepository.findById(ingredient.getId())
+                                                           .orElse(new Ingredient());
+        BigDecimal persistPrice = persistIngredient.getPrice();
+        persistIngredient.setPrice(persistPrice.add(ingredient.getAmount()));
```

Comment:
[major] This definitely does something weird. But as this is mostly for Stage 2 you can do nothing here for this PR.